### DISCUSSION TITLE
Encoding NEXT_DATA json

### DISCUSSION THREAD

**Discussion by PiotrSzlagura** [vercel/next.js#41433] [Ideas]
### Describe the feature you'd like to request

Is it possible to introduce (or implement it for given project, I would be 100% satisfied with a guide how to implement it myself) a function, that will encode NEXT_DATA into base64 or something like that and decode it when needed?

The problem which this function can fix, is Google crawling bits of NEXT_DATA and trying to open links which are not valid (in my case, I use parts of urls in next data with which I build urls in code, Google then tries to enter that url part and reports a lot of 404's).

### Describe the solution you'd like

1. During `getInitialProps`, encode data in BASE64 (or other format)
2. When hydrating on client side, decode this data and use it as it would be used otherwise

### Describe alternatives you've considered

I tried adding something like `rel="nofollow"` and looked into workarounds like `<!--googleoff: all-->` (which doesnt work).

One other idea - would it be possible to create a special variant of page (only for robots) which would have no NEXT_DATA script at all? This would also solve my primary issue

**Comment by PiotrSzlagura** [comment]
There is one more thing which troubles me with current state of NEXT_DATA - on each page this JSON contains data about current page, this string starts with `/` and is crawled by Google (but this is not a real url, it's something like `[locale]/companies`. This negatively affects SEO

**Comment by c0b41** [comment]
did you find any solution?

**Reply by PiotrSzlagura** [reply]
We might say that I did, but I wouldn't call that a solution.

I finally decided to make sure that my NEXT_DATA doesn't contain any strings starting with `/`...

As for the problem with url patterns being picked up by Google - I still don't have any solution for it unfortunately

**Comment by olaj** [comment]
We have the same issue. Would be nice to be able to encoding. But i will also look into trying not have anything starting with / in there.
